packages/twenty-server/src/engine/metadata-modules/object-metadata/object-metadata.service.ts

```diff
@@ -212,6 +219,18 @@ export class ObjectMetadataService extends TypeOrmQueryService<ObjectMetadataEnt
     input: UpdateOneObjectInput,
     workspaceId: string,
   ): Promise<ObjectMetadataEntity> {
+    const { update } = input;
+
+    if (isDefined(update.labelSingular)) {
+      update.labelSingular = capitalize(update.labelSingular);
```

Comment:
I'm not a fan of modifying the original object directly, this could lead to unwanted side effects.

Let's take the following example:

```
function save(input) {
 input.update = 'toto'
}

main() {
const input = { toto: 'tata' }
save(input)

print(input) // input has been changed which is a side effect of save but I don't expect it from the outside, this can lead to bugs)
```

Reply:
Instead, I would direclty change line 228:

```
const existingObjectMetadataCombinedWithUpdateInput = {
      ...existingObjectMetadata,
      ...input.update,
      ...{ isDefined(update.labelSingular) ? capitalize(update.labelSingular) : undefined },
      ...{ isDefined(update.labelPlural) ? capitalize(update.labelPlural) : undefined },
    };
```

If you don't like this syntax, I would introduce an intermediary function: capitalizeLabelsInInput(input) that would return a new js object